src/System Application/App/Extension Management/src/ExtensionInstallationImpl.Codeunit.al

```diff
@@ -364,5 +364,41 @@ codeunit 2500 "Extension Installation Impl"
         ExtensionDetails.Run();
         exit(ExtensionDetails.Editable());
     end;
+
+    internal procedure AllowsDebug(ResourceProtectionPolicy: Integer): Boolean
+    begin
+        exit(HasPolicy(ResourceProtectionPolicy, 1));
+    end;
+
+    internal procedure AllowsDownloadSource(ResourceProtectionPolicy: Integer): Boolean
+    begin
+        exit(HasPolicy(ResourceProtectionPolicy, 2));
+    end;
+
+    internal procedure AllowsDownloadSourceInSymbols(ResourceProtectionPolicy: Integer): Boolean
+    begin
+        exit(HasPolicy(ResourceProtectionPolicy, 4));
+    end;
+
+    local procedure HasPolicy(ResourceProtectionPolicy: Integer; PolicyToTestFor: Integer): Boolean
+    var
+        remaining, tester, result, nextFlag : integer;
```

Comment:
Variable names casing, also integer -> Integer

Reply:
remains one of the hardest things about AL code